cartographer/EncodeDetect.h

```diff
@@ -0,0 +1,9 @@
+
```

Comment:
A few things on this file:
1. add `#pragma once` on top
2. the structs on 8/9 should be defined as external and actually defined in the cpp
3. Why is there iostream here?

Reply:
1.) done
2.) not needed
3.) Removed